Update requirements.txt

wiljav opened the pull request:
Changed `torch` and `gradio` versions to install latest version instead on one specific version.

The previous version caused issues related to dependency conflicts, [example here](https://github.com/TonyLianLong/stable-diffusion-xl-demo/issues/9):

wiljav committed:
Update requirements.txt

Changed 'torch' and 'gradio' versions to install latest version instead on one specific version

TonyLianLong commented:
Have you tested the demo and made sure the style (CSS) does not break? If so, I'll merge the PR.

wiljav committed:
Update requirements.txt

Added 'peft' and changed back 'gradio' to the old version 3.14.0, because there are some outdated syntax within app.py that the new gradio version (> 4.0) doesn't recognise them.

wiljav commented:
I added `peft` and changed back `gradio` to version 3.14.0, the new version (>4.0) doesn't support some deprecated syntax (https://www.gradio.app/changelog#4-21-0) in app.py, which caused issues.
Tested and works as expected now.
